--- a/src/http/ngx_http_upstream.c	2023-09-12 11:09:36.503713749 +0200
+++ b/src/http/ngx_http_upstream.c	2023-04-11 03:45:34.000000000 +0200
@@ -620,8 +620,10 @@
         r->write_event_handler = ngx_http_upstream_wr_check_broken_connection;
     }
 
-    if (r->request_body) {
-        u->request_bufs = r->request_body->bufs;
+    if (r->method != NGX_HTTP_PUT) {                                                                                                                               
+     if (r->request_body) {                                                                                                                                      
+        u->request_bufs = r->request_body->bufs;                                                                                                               
+     } 
     }
 
     if (u->create_request(r) != NGX_OK) {
